pkgs/misc/vim-plugins/patches/coc-nvim/nodejs-path.patch

```diff
@@ -0,0 +1,39 @@
+diff --git a/autoload/coc/util.vim b/autoload/coc/util.vim
+index 5c4f4bb1..6d90bb1c 100644
+--- a/autoload/coc/util.vim
++++ b/autoload/coc/util.vim
+@@ -325,7 +325,7 @@ function! coc#util#job_command()
+   if (has_key(g:, 'coc_node_path'))
+     let node = expand(g:coc_node_path)
+   else
+-    let node = $COC_NODE_PATH == '' ? 'node' : $COC_NODE_PATH
++    let node = '@nodejs@' == '' ? 'node' : '@nodejs@'
```

Comment:
```suggestion
+    let node = '$COC_NODE_PATH' == '' ? '@nodejs@' : $COC_NODE_PATH
```

Don't we rather want to override the default nodejs location, and still allow overriding it with `$COC_NODE_PATH`?

Reply:
you could export  COC_NODE_PATH or ask coc to default to `g:node_host_prog` that is set in neovim wrapper ?

Reply:
> Don't we rather want to override the default nodejs location, and still allow overriding it with `$COC_NODE_PATH`?

My thought here is: If you're building with nix you can simply override `nodejs` with any node version, `$COC_NODE_PATH` is then redundant, and being able to configure this through both nix and environment variables can lead to hard to debug configuration issues.  I'm not sure what the precedence is here, and I'm more than happy to defer to your judgement.  Just let me know.

Reply:
> Don't we rather want to override the default nodejs location, and still allow overriding it with `$COC_NODE_PATH`?

I'm happy to defer to you here.  I wasn't sure what the idiomatic way to do this was, my thinking was that if someone configured a path in nix they wouldn't then want to override it.

Reply:
> you could export COC_NODE_PATH or ask coc to default to `g:node_host_prog` that is set in neovim wrapper ?

1. Setting COC_NODE_PATH would be ideal if executed directly but since it's a vim plugin I'm not sure whether it would be desirable to export `COC_NODE_PATH` to neovim.  Is there a way to do this cleanly?

2. Defaulting to g:node_host_prog may make sense but since that's not nix specific functionality doesn't that seems like something that should be handled upstream?